### DISCUSSION TITLE
How to make stick combo

### DISCUSSION THREAD

**Discussion by BrandoJotaro** [ndeadly/MissionControl#427] [Q&A]
Hi! How could I make a button combo which triggers the left stick to be pressed down.

This would be the concept:

if (buttons->X) {
    buttons->ZL= 1;
    Left stick->downwards = 1
}

**Comment by ndeadly** [comment]
You can set the stick state through the [`SwitchAnalogStick`](https://github.com/ndeadly/MissionControl/blob/master/mc_mitm/source/controllers/switch_analog_stick.hpp) class.

This data is accessible via the [incoming report](https://github.com/ndeadly/MissionControl/blob/0ee33b355b2d8649dddcb754ca928d15778636de/mc_mitm/source/controllers/switch_controller.hpp#L257) (official controllers), or a [member](https://github.com/ndeadly/MissionControl/blob/0ee33b355b2d8649dddcb754ca928d15778636de/mc_mitm/source/controllers/emulated_switch_controller.hpp#L76-L77) of `EmulatedSwitchController` (unoffical controllers)

Use the `SetData` method to set the stick coordinates to as you wish. For example (you might need to replace `UINT12_MAX` with 0, I can't remember whether the y-axis is inverted or not)

`switch_report->input0x30.left_stick.SetData(STICK_ZERO, UINT12_MAX)`
`m_left_stick.SetData(STICK_ZERO, UINT12_MAX);`

The `ApplyButtonCombos` method only takes the button state as an argument, so you would need to modify this to take the stick values too if you want to add your combos in that location.
